.github/workflows/build-test.yml

```diff
@@ -28,6 +28,7 @@ jobs:
     # Archive the outputs
     - name: 'Archive Logs'
       uses: actions/upload-artifact@v2
+      if: ${{ always() }}
```

Comment:
Technically the `${{` and `}}` here are optional

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#about-contexts-and-expressions

> When you use expressions in an `if` conditional, you may omit the expression syntax (`${{ }}`) because GitHub automatically evaluates the `if` conditional as an expression, unless the expression contains any operators. If the expression contains any operators, the expression must be contained within `${{ }}` to explicitly mark it for evaluation. For more information about `if` conditionals, see "Workflow syntax for GitHub Actions."

And yes, the fact that their example uses them is unfortunate.

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#example-7